mne/utils/config.py

```diff
@@ -398,8 +450,26 @@ def set_config(key, value, home_dir=None, set_env=True):
     directory = op.dirname(config_path)
     if not op.isdir(directory):
         os.mkdir(directory)
-    with open(config_path, "w") as fid:
-        json.dump(config, fid, sort_keys=True, indent=0)
+
+    # Adapting the mode depend if you are create the file
+    # or no.
+    mode = "r+" if op.isfile(config_path) else "w+"
+
+    with _open_lock(config_path, mode) as fid:
+        try:
+            fid.seek(0)
```

Comment:
Why `fid.seek(0)` here? It should start at position 0 when you open it, no?

Reply:
you are rigth!